From d7c41e5239dac250a15d54d9b9f7f39d060f76cd Mon Sep 17 00:00:00 2001
From: Kunkun Jiang <jiangkunkun@huawei.com>
Date: Thu, 13 Oct 2022 14:51:08 +0800
Subject: [PATCH] iommu: Fix compliation failure caused by
 iommu_device_register
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: d7c41e5239dac250a15d54d9b9f7f39d060f76cd
Modified-by-SEL: No


virt inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5UN39
CVE: NA

------------------------------

Two redundant arguements are added to 'iommu_device_register' of
'include/linux/iommu.h' in the commit bbf3b39e5576 ("iommu:
Introduce dirty log tracking framework"). As a result, compiling
the kernel fails when the CONFIG_IOMMU_API is disabled. Delete the
two redundant arguements to solve this problem.

Fixes: bbf3b39e5576 ("iommu: Introduce dirty log tracking framework")
Reported-by: Yejian Zheng <zhengyejian1@huawei.com>
Signed-off-by: Kunkun Jiang <jiangkunkun@huawei.com>
Reviewed-by: Keqian Zhu <zhukeqian1@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/iommu.h | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/include/linux/iommu.h b/include/linux/iommu.h
index 092384b71ab2..d993036c94c2 100644
--- a/include/linux/iommu.h
+++ b/include/linux/iommu.h
@@ -1031,9 +1031,7 @@ static inline int iommu_clear_dirty_log(struct iommu_domain *domain,
 	return -EINVAL;
 }
 
-static inline int iommu_device_register(struct iommu_device *iommu,
-					const struct iommu_ops *ops,
-					struct device *hwdev)
+static inline int iommu_device_register(struct iommu_device *iommu)
 {
 	return -ENODEV;
 }
-- 
2.34.1

